src/libstd/sys/windows/mutex.rs

```diff
@@ -58,6 +58,8 @@ pub unsafe fn raw(m: &Mutex) -> c::PSRWLOCK {
 impl Mutex {
     pub const fn new() -> Mutex {
         Mutex {
+            // This works because SRWLOCK_INIT is a NULL pointer, so we are also properly
```

Comment:
Technically `SRWLOCK` is a struct wrapping a pointer which isn't actually used as a pointer.

Reply:
Oh wow. I guess Windows API doesn't really care about meaningful types?

Reply:
Fixed, I hope.